src/Neo/SmartContract/StorageKey.cs

```diff
@@ -67,7 +74,9 @@ public bool Equals(StorageKey other)
 
         public override int GetHashCode()
         {
-            return Id + (int)Key.Span.Murmur32(0);
+            if (_hashCode == 0)
```

Comment:
Why just when 0?

Reply:
> Why just when 0?

Non-zero means it has already been calculated.